src/components/icons/LottieAnimation.tsx

```diff
@@ -1,36 +1,125 @@
 import Player, { LottieRefCurrentProps } from 'lottie-react'
-import React, { useEffect, useRef } from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 
 interface LottieAnimationProps {
   animationData: any
   speed?: number
+  color?: string
   onComplete?: () => void
   style?: React.CSSProperties
 }
 
+/**
+ * Converts hex color to RGBA.
+ *
+ * @param hex - The hex color code.
+ * @returns The RGBA values.
+ */
+const hexToRGBA = (hex: string): number[] => {
+  let r = 0,
+    g = 0,
+    b = 0
+  const a = 1
+  if (hex.length === 7) {
+    r = parseInt(hex.slice(1, 3), 16) / 255
+    g = parseInt(hex.slice(3, 5), 16) / 255
+    b = parseInt(hex.slice(5, 7), 16) / 255
+  }
+  return [r, g, b, a]
+}
+
+/**
+ * Updates colors in a single shape item.
+ *
+ * This function checks if the item represents a stroke or
+ * fill and updates its color using the provided RGBA array.
+ * If the item contains a nested items array, it recursively
+ * calls itself to ensure that all relevant shapes within the
+ * structure are updated with the new color.
+ *
+ * @param item - The shape item object which may contain
+ * stroke or fill color properties.
+ * @param rgbaArray - The RGBA values to set as new color.
+ */
+const updateColorsInItem = (item: any, rgbaArray: number[]) => {
+  if ((item.ty === 'st' || item.ty === 'fl') && item.c) {
+    // Check for animated color property
+    if (item.c.a === 1) {
+      item.c.k.forEach((keyframe: any) => {
+        keyframe.s = rgbaArray
+      })
+    } else {
+      item.c.k = rgbaArray
```

Comment:
Please note that it is dangerous and considered a bad practice to mutate arguments. Arguments are the input to a function and should generally not be modified. Use the return value as the output. Pure functions are easier to reason about and more self-contained. Modifying the argument has upstream implications and can affect anything above it in the call stack.

I won't make you rewrite the function this time, but please keep this recommendation in mind for the future. Thanks!

Reply:
Thank you for highlighting the importance of functional purity and immutability. I completely agree that avoiding the mutation of arguments is crucial for maintaining code readability and reliability. It ensures that functions remain predictable and easier to test. 
However, I also considered that copying a large object, like the JSON animation data, might significantly affect performance. 
To address this, I enhanced the handling of Lottie animation data by implementing immutability and memoization strategies.

Reply:
Yes, I think memoization is the right call to ensure good performance here!

Reply:
I'd slightly prefer to use something like `map` to avoid mutating within `updateColorsInItem` at all. That said, I can confirm that this is only used by `changeLineColor` which passes a clone, and the result is now memoized with `useMemo`. Thank you, @ohrytskov!